pymongo/uri_parser.py

```diff
@@ -514,6 +519,9 @@ def parse_uri(uri, default_port=DEFAULT_PORT, validate=True, warn=False,
                     options[opt] = val
         if "tls" not in options and "ssl" not in options:
             options["tls"] = True if validate else 'true'
+    #elif not is_srv and options.get("srvServiceName", None) is not None:
```

Comment:
Remove the dead code.

Reply:
This isn't dead code, it was to fix the test that was testing for a configuration to be raised under those conditions. So I restored it.